### DISCUSSION TITLE
Hot Reload .net 6?

### DISCUSSION THREAD

**Discussion by dotnetshadow** [fullstackhero/blazor-starter-kit#362] [Q&A] [answered]
Hi all,

With .net 6 has anyone managed to get hot reload working? I've change the references for the project to .net 6 but can't seem to get the hot reload working. Any suggestions?

**Comment by dotnetshadow** [comment, marked as answer]
I had to update the nuget packages and it seems hot reload is working

**Reply by GLARDEN** [reply]
No luck for me on Visual Studio 2022, won't even hot reload a hello world console app, even when I run dotnet watch. 
I'm starting to wonder if it is a permission issue or maybe because I have multiple instances installed.

**Reply by dotnetshadow** [reply]
Maybe you haven't got one of the workloads installed?

**Reply by chhinsras** [reply]
> Maybe you haven't got one of the workloads installed?

Which workload ,

**Reply by GLARDEN** [reply]
I suspect you may be onto something!! I will install everything and report back on the results!

**Reply by GLARDEN** [reply]
I added "Desktop development with C++", "Azure development",".net desktop development" and voila!!! Hot load is working in a default blazor wazm project!!!! Thanks for the suggestion.. I'm going to try my actual solution and see if it works!

**Reply by GLARDEN** [reply]
I take it back.. it was working and has stopped again. When it worked on my newly created blazor wasm template , it was very fast! I added a form and textbox and it worked. I created a compile error and it stopped working and hasn't worked since.. Now it won't work on even a simple console app again. So right now it's very spotty, but when they get it working it's going to be awesome!

**Reply by rwobig93** [reply]
I also haven't been able to get hot reload working on my existing projects, I tried following the Microsoft migration guide, manually moved everything over to a new WASM project and continue to experience no hot reload *goodness*.
So after finding no others reporting the generic-ness of "hot reload not working" I decided to try to use dotnet watch from the CLI directly and found that when a file updates dotnet watch just reports on the CLI that the file changed w/ nothing else.
If I then Ctrl + C I get an unhandled exception, so using that I found others are experiencing this same issue migrating existing projects over as well here:
https://github.com/dotnet/aspnetcore/issues/37462

This is fairly frustrating for sure and I haven't found the fix yet but wanted to post this here in case someone else stumbles upon it, if I find the fix I'll report back but if someone else finds a fix please also let me know before I descend into further madness :)

**Reply by rwobig93** [reply]
I ended up finding a solution by forcing the dotnet watch script into index.html and setting VS 2022 to launch dotnet watch w/o run in the launchSettings.json and this workaround is now working (doesn't launch the browser automatically but that's a small tradeoff for me), details are on a response to the bug report:
https://github.com/dotnet/aspnetcore/issues/37462#issuecomment-968329005
